mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -709,11 +713,12 @@ inline unsigned obtainGenericTensorTransformationInfo(
   // 3. For memrefs with embedded maps, use its input rank.
   assert(type.isa<MemRefType>() || type.isa<VectorType>());
   unsigned coordLength = 0;
-  ArrayRef<AffineMap> typeAffineMaps;
+  SmallVector<AffineMap> typeAffineMaps;
   if (type.isa<MemRefType>()) {
     MemRefType memrefType = type.cast<MemRefType>();
     coordLength = memrefType.getRank();
-    typeAffineMaps = memrefType.getAffineMaps();
+    if (!memrefType.getLayout().isIdentity())
```

Comment:
An identity map does nothing, and is the new way to represent what was an empty map list before.